base/complex.jl

```diff
@@ -56,11 +56,29 @@ Return both the real and imaginary parts of the complex number `z`.
 """
 reim(z) = (real(z), imag(z))
 
-real{T<:Real}(::Type{T}) = T
-real{T<:Real}(::Type{Complex{T}}) = T
+"""
+    real{R<:Real}(::Type{R})
```

Comment:
In fact, we should probably define a fallback `real(T::Type) = typeof(real(zero(T)))`.